analysis_service/base/forms.py

```diff
@@ -0,0 +1,448 @@
+                    filesizeformat(uploaded_file.size)
+                )
+            )
+        contents = uploaded_file.read()
+        if not contents.startswith('ssh-rsa AAAAB3'):
+            raise ValidationError(
+                'Invalid public key (a public key should start with \'ssh-rsa AAAAB3\')'
+            )
+        return contents
+
+
+class ClusterIdField(forms.IntegerField):
+    def __init__(self, *args, **kwargs):
+        super(ClusterIdField, self).__init__(*args, **kwargs)
+
+    def clean(self, data):
+        cluster_id = super(ClusterIdField, self).clean(data)
+        if not models.Cluster.objects.filter(id=cluster_id).exists():
+            raise ValidationError('Cluster {} not found'.format(cluster_id))
+        return cluster_id
+
+
+class ScheduledSparkIdField(forms.IntegerField):
+    def __init__(self, *args, **kwargs):
+        super(ScheduledSparkIdField, self).__init__(*args, **kwargs)
+
+    def clean(self, data):
+        scheduled_spark_id = super(ScheduledSparkIdField, self).clean(data)
+        if not models.ScheduledSpark.objects.filter(id=scheduled_spark_id).exists():
+            raise ValidationError('Scheduled Spark job {} not found'.format(scheduled_spark_id))
+        return scheduled_spark_id
+
+
+class NewClusterForm(forms.ModelForm):
+    identifier = forms.RegexField(
+        required=True,
+        regex="^[\w-]{1,100}$",
+        widget=forms.TextInput(attrs={
+            'class': 'form-control',
+            'pattern': r'[\w-]+',
+            'data-toggle': 'popover',
+            'data-trigger': 'focus',
+            'data-placement': 'top',
+            'data-container': 'body',
+            'data-content': 'A brief description of the cluster\'s purpose, '
+                            'visible in the AWS management console.',
+            'data-validation-pattern-message': 'Valid cluster names are strings of alphanumeric '
+                                               'characters, \'_\', and \'-\'.',
+        })
+    )
+    size = forms.IntegerField(
+        required=True,
+        min_value=1, max_value=20,
+        widget=forms.NumberInput(attrs={
+            'class': 'form-control',
+            'required': 'required',
+            'min': '1', 'max': '20',
+            'data-toggle': 'popover',
+            'data-trigger': 'focus',
+            'data-placement': 'top',
+            'data-container': 'body',
+            'data-content': 'Number of workers to use in the cluster '
+                            '(1 is recommended for testing or development).',
+        })
+    )
+    public_key = PublicKeyFileField(
+        required=True,
+        widget=forms.FileInput(attrs={'required': 'required'})
+    )
+
+    def save(self, user):
+        # create the model without committing, since we haven't
+        # set the required created_by field yet
+        new_cluster = super(NewClusterForm, self).save(commit=False)
```

Comment:
This is a bit confusing, I think you can make it simpler by overriding the __init__ method of NewClusterForm:
```python
def __init__(self, user, *args, **kwargs):
    self.created_by = user
    super(NewClusterForm, self).__init__(*args, **kwargs)
```
and then you can pass request.user directly to the NewClusterForm constructor.